bundles/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/internal/WakeOnLanUtility.java

```diff
@@ -0,0 +1,160 @@
+
+    /**
+     * Get MAC address for host
+     * uses "arp" tool
+     *
+     * @param hostName Host Name (or IP address) of host to retrieve MAC address for
+     * @return MAC address
+     */
+    public static @Nullable String getMACAddress(String hostName) {
+        try {
+            Process proc = Runtime.getRuntime().exec("arping -r -c 1 -C 1 " + hostName); // arp on linux
+            int returnCode = proc.waitFor();
+            String s;
+            StringBuilder builder = new StringBuilder();
+            BufferedReader input = new BufferedReader(
+                    new InputStreamReader(returnCode == 0 ? proc.getInputStream() : proc.getErrorStream()));
+            while ((s = input.readLine()) != null) {
+                builder.append(s);
+            }
+
+            if (returnCode != 0) {
+                logger.debug("getMacAddress error stream: {}", builder.toString());
+            } else {
+
+                Pattern macPattern = Pattern.compile("(([0-9a-fA-F]{2}[:-]){5}[0-9a-fA-F]{2})");
+                Matcher matcher = macPattern.matcher(builder.toString());
+                String macAddress = null;
+
+                while (matcher.find()) {
+                    String group = matcher.group();
+
+                    if (group.length() == 17) {
+                        macAddress = group;
+                    }
+                }
+
+                if (macAddress != null) {
+                    logger.debug("MAC address of host {} is {}", hostName, macAddress);
+                    return macAddress;
+                }
+            }
+        } catch (IOException | InterruptedException e) {
+            logger.debug("Problem getting MAC address: {}", e.getMessage());
+        }
+        return null;
+    }
+
+    /**
+     * Send single WOL (Wake On Lan) package on all interfaces
+     *
+     * @macAddress MAC address to send WOL package to
+     */
+    public static void sendWOLPacket(String macAddress) {
+        byte[] bytes = getWOLPackage(macAddress);
+
+        try {
+            Enumeration<NetworkInterface> interfaces = NetworkInterface.getNetworkInterfaces();
+            while (interfaces.hasMoreElements()) {
+                NetworkInterface networkInterface = interfaces.nextElement();
+                if (networkInterface.isLoopback()) {
+                    continue; // Do not want to use the loopback interface.
+                }
+                for (InterfaceAddress interfaceAddress : networkInterface.getInterfaceAddresses()) {
+                    InetAddress broadcast = interfaceAddress.getBroadcast();
+                    if (broadcast == null) {
+                        continue;
+                    }
+
+                    try {
+                        DatagramPacket packet = new DatagramPacket(bytes, bytes.length, broadcast, 9);
+                        DatagramSocket socket = new DatagramSocket();
+                        socket.send(packet);
+                        socket.close();
```

Comment:
Please wrap socket in a try-with-resources

Reply:
done